### PR TITLE
Use ocamlformat

### PR DESCRIPTION
**Pull request by Leonidas-from-XIV**
I've added the config that Jane Street uses which is a style we often tend to follow anyway.

### CODE REVIEW

File: lib/ppx_mysql_runtime.ml [hunk 1]
```diff
@@ -1,14 +1,10 @@
 (********************************************************************************)
 (** {1 Public functions and values}                                             *)
+
 (********************************************************************************)
 
 let identity x = x
+let map_option f = function Some x -> Some (f x) | None -> None
+let get_option = function Some x -> x | None -> assert false
```
Comment (AndreasDahl) [inline lib/ppx_mysql_runtime.ml]: I don't like this compared to what was before.

File: lib/ppx_mysql_runtime.ml [hunk 2]
```diff
@@ -1,14 +1,10 @@
...
 
-let map_option f = function
-    | Some x -> Some (f x)
-    | None   -> None
-
-let get_option = function
-    | Some x -> x
-    | None   -> assert false (* FIXME *)
-
+(* FIXME *)
```
Comment (AndreasDahl) [inline lib/ppx_mysql_runtime.ml]: It kinda screwed with this comment. Completely out of place now.

File: lib/ppx_mysql_runtime.mli [hunk 3]
```diff
@@ -1,10 +1,9 @@
...
-(** {1 Public functions and values}                                             *)
-(********************************************************************************)
-
-val identity: 'a -> 'a
 
-val map_option: ('a -> 'b) -> 'a option -> 'b option
+(** {1 Public functions and values}                                             *)
 
-val get_option: 'a option -> 'a
+(********************************************************************************)
```
Comment (AndreasDahl) [inline lib/ppx_mysql_runtime.mli]: It concerns me a little that our formatter would be so concerned about pure comments.

File: ppx/ppx_mysql.ml [hunk 4]
```diff
@@ -1,268 +1,280 @@
...
-    opt: bool;
-    name: string;
-    of_string: string * string;
-    to_string: string * string;
-    }
+  { typ : string
+  ; opt : bool
+  ; name : string
+  ; of_string : string * string
+  ; to_string : string * string }
```
Comment (AndreasDahl) [inline ppx/ppx_mysql.ml]: Weird, but i can live with this.

File: ppx/ppx_mysql.ml [hunk 5]
```diff
@@ -1,268 +1,280 @@
...
-                Prepared.fetch stmt_result >>= function
-                    | Some _ -> IO.return (Error `Expected_none_found_one)
-                    | None   -> IO.return (Ok ())
-            ]
-        | other ->
-            raise (Location.Error (Location.Error.createf ~loc "Error in 'mysql' extension: I don't understand query variant '%s'" other)) in
-    match parse_query query with
-        | Ok {query; in_params; out_params} ->
-            (* Note that in the expr fragment below we disable warning 26 (about unused variables)
+    (* FIXME *)]
```
Comment (AndreasDahl) [inline ppx/ppx_mysql.ml]: hmmm...

File: lib/ppx_mysql_runtime.ml (hunk 2 above)
Comment (Leonidas-from-XIV) [reply lib/ppx_mysql_runtime.ml]: Yeah, but also I absolutely hate in-line comments. I would like a language which made this a syntax error.

File: lib/ppx_mysql_runtime.mli (hunk 3 above)
Comment (Leonidas-from-XIV) [reply lib/ppx_mysql_runtime.mli]: But it sort of isn't, because these are comments for the documentation generator.

File: ppx/ppx_mysql.ml (hunk 4 above)
Comment (Leonidas-from-XIV) [reply ppx/ppx_mysql.ml]: I agree. Found no option to change it though. If this turns out to be the community consensus I guess I can live with it too.

File: lib/ppx_mysql_runtime.ml (hunk 1 above)
Comment (Leonidas-from-XIV) [reply lib/ppx_mysql_runtime.ml]: I agree. I made it break all pattern matches. Maybe it is better now?

File: examples/mysql_with_identity_monad.ml [hunk 6]
```diff
@@ -91,5 +106,7 @@ let () =
     get_user dbh ~id:2l >>= fun user -> print_user user; Mysql.disconnect dbh; Ok ()
   in
   match result with
-  | Ok () -> print_endline "All went well!"
-  | Error _ -> print_endline "An error occurred!"
+  | Ok () ->
+      print_endline "All went well!"
```
Comment (AndreasDahl) [inline examples/mysql_with_identity_monad.ml]: It's a shame that each case can't be one line. Still prefer this to multiple cases on the same line

File: ppx/ppx_mysql.ml [hunk 7]
```diff
@@ -66,23 +72,31 @@ let parse_query =
...
               ; out_params = List.rev acc_out }
-        | Some _ -> Error `Unterminated_string
+        | Some _ ->
+            Error `Unterminated_string
       else
         let this = query.[i] in
         match string_delim with
-        | _ when Char.(this = '\\') ->
+        | _
+          when Char.(this = '\\') ->
```
Comment (AndreasDahl) [inline ppx/ppx_mysql.ml]: Would prefer `when` cases to be on the same line (assuming it fits). But this is OK

Comment (Leonidas-from-XIV) [reply ppx/ppx_mysql.ml]: I would like a `break-guards = fit` setting, agreed. Maybe I should file a feature request?

File: examples/mysql_with_identity_monad.ml (hunk 6 above)
Comment (Leonidas-from-XIV) [reply examples/mysql_with_identity_monad.ml]: Guess ocamlformat opts for

> foolish consistency is the hobgoblin of the little minds

**Comment by Leonidas-from-XIV** [comment]
So, should we go with this?

**Pull request merged**